templates/defineOperation.dot

```diff
@@ -1,13 +1,19 @@
 {-| {{= it.doc }}
-{{? it.input && it.input.members }}
-__Parameters__
+{{? it.requiredParams }}
+__Required Parameters__
 
-{{~ it.input.members :m }}* `{{= m.key }}` __:__ `{{= m.value.type }}`
+{{~ it.requiredParams :p }}* `{{= p.key }}` __:__ `{{= p.value.type }}`
 {{~}}
 {{?}}
 -}
-{{= it.name }} : {{? it.input }}{{~ it.input.members :m }}{{= m.value.type }} -> {{~}}{{?}}AWS.ServiceConfig -> AWS.Http.UnsignedRequest {{= it.output.type }}
-{{= it.name }} {{? it.input }}{{~ it.input.members :m }}{{= m.key }} {{~}}{{?}}serviceConfig =
+{{= it.name }} :
+    {{~ it.requiredParams :p }}{{= p.value.type }}
+    -> {{~}}{{? it.optionalParams.length }}({{= it.optionsName }} -> {{= it.optionsName }})
+    -> {{?}}AWS.Http.UnsignedRequest {{= it.output.type }}
+{{= it.name }} {{~ it.requiredParams :p }}{{= p.key }} {{~}}{{? it.optionalParams.length }}setOptions {{?}}={{? it.optionalParams.length }}
+  let
+    options = setOptions ({{= it.optionsName }} {{= it.optionalParams.map(() => 'Nothing').join(' ') }})
```

Comment:
I don't think this is possible. The given implementation is ugly, but works, and is hidden from the client